semtypes/src/main/java/io/ballerina/semtype/subtypedata/StringSubtype.java

```diff
@@ -17,19 +17,24 @@
  */
 package io.ballerina.semtype.subtypedata;
 
+import io.ballerina.semtype.EnumerableString;
+import io.ballerina.semtype.EnumerableSubtype;
 import io.ballerina.semtype.ProperSubtypeData;
 import io.ballerina.semtype.SubtypeData;
 
+import java.util.Collections;
+import java.util.List;
+
 /**
  * Represent StringSubtype.
  *
  * @since 2.0.0
  */
-public class StringSubtype implements ProperSubtypeData {
+public class StringSubtype extends EnumerableSubtype implements ProperSubtypeData {
     public final boolean allowed;
-    public final String[] values;
+    public final List<EnumerableString> values;
```

Comment:
Why is this a list?

Reply:
I noticed that FloatSubtype also have a list, shall we convert that to a array too.
The reason to make these things arrays is to minimize the memory usage as we don't do any list operations with these (list operations as in adding things to this list)

Reply:
can we maintain the immutability in `createStringSubtype` and `createFloatSubtype` with an array (Or is the readonly state optional)?

Reply:
We can't; but I don't think we need to, as we are just porting the Ballerina version and the Ballerina version maintain the statically typed readonly-ness. And addition to the sem-type will be done in Ballerina version and then be back ported to Java.

Reply:
changed float and String subtypes

Reply:
might need to do changes in `EnumerableSubtype` as well

Reply:
I realized that there is allowed and values fields in both the parent class `EnumerableSubtype` and subclasses. Shall I keep only the fields in the super class and cast when needed?